File: app/src/main/java/dolphin/android/apps/BloodServiceApp/ui/SplashActivity.java
package dolphin.android.apps.BloodServiceApp.ui;

import android.app.Activity;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Build;
import android.os.Bundle;
import android.os.LocaleList;
import android.widget.TextView;

import com.google.android.gms.common.ConnectionResult;
import com.google.android.gms.common.GoogleApiAvailability;
import com.google.firebase.remoteconfig.FirebaseRemoteConfig;

import java.util.Locale;

import dolphin.android.apps.BloodServiceApp.R;


public class SplashActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Configuration config = getBaseContext().getResources().getConfiguration();
        //set default locale
        //http://stackoverflow.com/a/4239680
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            LocaleList list = new LocaleList(Locale.TAIWAN);
            LocaleList.setDefault(list);
            config.setLocales(list);
//        } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
//            Configuration configuration = new Configuration();
//            configuration.setLocale(Locale.TAIWAN);
//            applyOverrideConfiguration(configuration);
        } else {
            config.locale = Locale.TAIWAN;
            Locale.setDefault(config.locale);
            getBaseContext().getResources().updateConfiguration(config,
                    getBaseContext().getResources().getDisplayMetrics());
        }

        setContentView(R.layout.activity_splash);

        //FirebaseAnalytics.getInstance(this);//initialize this
        FirebaseRemoteConfig.getInstance();

        checkGoogleApiAvailability();
    }

    private void checkGoogleApiAvailability() {
        //http://stackoverflow.com/a/31016761/2673859
        //check google play service and authentication
        GoogleApiAvailability googleAPI = GoogleApiAvailability.getInstance();
        int result = googleAPI.isGooglePlayServicesAvailable(this);
        if (result != ConnectionResult.SUCCESS) {
            //Log.e(TAG, googleAPI.getErrorString(result));
            TextView textView = (TextView) findViewById(android.R.id.message);
            if (textView != null) {
                textView.setText(googleAPI.getErrorString(result));
            }
            googleAPI.getErrorDialog(this, result, 0, new DialogInterface.OnCancelListener() {
                @Override
                public void onCancel(DialogInterface dialogInterface) {
                    finish();
                }
            }).show();
            return;//don't show progress bar
        }

        Intent intent = new Intent(this, MainActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK);
        overridePendingTransition(0, 0);
        startActivity(intent);
        overridePendingTransition(0, 0);
        finish();
        overridePendingTransition(0, 0);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        //super.onActivityResult(requestCode, resultCode, data);
        checkGoogleApiAvailability();
    }
}

File: app/src/main/java/dolphin/android/apps/BloodServiceApp/ui/SplashActivity.kt
package dolphin.android.apps.BloodServiceApp.ui

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import com.google.android.gms.common.ConnectionResult
import com.google.android.gms.common.GoogleApiAvailability
import com.google.firebase.remoteconfig.FirebaseRemoteConfig
import dolphin.android.apps.BloodServiceApp.MyApplication
import dolphin.android.apps.BloodServiceApp.R
import dolphin.android.apps.BloodServiceApp.pref.PrefsUtil
import dolphin.android.apps.BloodServiceApp.provider.LocaleUtil


class SplashActivity : Activity() {
    override fun attachBaseContext(newBase: Context?) {
        super.attachBaseContext(LocaleUtil.onAttach(newBase!!))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_splash)

        if (!checkGoogleApiAvailability()) {
            val myApp: MyApplication = application as MyApplication
            myApp.setGooglePlayServiceNotSupported()
            return
        }

        //FirebaseAnalytics.getInstance(this);//initialize this
        FirebaseRemoteConfig.getInstance()

        startMainActivity()
    }

    private fun checkGoogleApiAvailability(): Boolean {
        //http://stackoverflow.com/a/31016761/2673859
        //check google play service and authentication
        val googleAPI = GoogleApiAvailability.getInstance()
        val result = googleAPI.isGooglePlayServicesAvailable(this)
        if (result != ConnectionResult.SUCCESS) {
            //Log.e(TAG, googleAPI.getErrorString(result));
            val textView = findViewById<TextView>(android.R.id.message)
            textView.text = googleAPI.getErrorString(result)
            val dialog = googleAPI.getErrorDialog(this, result, 0)
            dialog.setOnDismissListener { startMainActivity() }
            dialog.show()
            return false//don't show progress bar
        }

        return true
    }

    private fun startMainActivity() {
        var intent = Intent(this, MainActivity2::class.java)
        if (!PrefsUtil.isUseActivity2(baseContext)) {//just a fallback, we usually do use this
            intent = Intent(this, MainActivity::class.java)
        }
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK
        overridePendingTransition(0, 0)
        startActivity(intent)
        overridePendingTransition(0, 0)
        finish()
        overridePendingTransition(0, 0)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        //super.onActivityResult(requestCode, resultCode, data);
        if (checkGoogleApiAvailability()) {
            startMainActivity()
        } else {
            finish()
        }
    }
}
